packages/editor/src/store/actions.js

```diff
@@ -67,8 +122,83 @@ export function* setupEditor( post, edits, template ) {
 		blocks = synchronizeBlocksWithTemplate( blocks, template );
 	}
 
-	yield resetEditorBlocks( blocks );
```

Comment:
I might run into it again in the future; I don't recall why I needed to reorder `resetEditorBlocks`, but it causes an issue where a new post will prompt about unsaved changes when `SCRIPT_DEBUG` is `true`, due to a fragile guarantee the current order establishes that causes "dirtiness" state to be unset by the `setupEditorState` action. This incidentally resolves the fact we dispatch this `setupEditor` action twice in an editing session, since it [happens in `constructor`](https://github.com/WordPress/gutenberg/blob/0d9efe27209c396ec6209622d58d706f2e8167c5/packages/editor/src/components/provider/index.js#L57) (rather than `componentDidMount`) and React helpfully detects side-effects via [double-invoking intended non-side-effect lifecycle functions](https://reactjs.org/docs/strict-mode.html#detecting-unexpected-side-effects).

Reply:
Oh, I remember now: We want the sourced attributes to be applied in `resetEditorBlocks`, which requires that we set the post state first (so that post meta can be read).

In that case, we might need to address the underlying issue with the `EditorProvider` lifecycle, or endure the prompts in `SCRIPT_DEBUG` (preferably not).

Reply:
From our conversation:

>So `resetEditorBlocks` sets the flag to true if the blocks are different.
`setupEditorState` sets it to false again.

>Basically it only worked because `setupEditorState` was called as last, so it reset the dirty flag

>The dirtying was actually pretty simple if we leave things as they are, and just call `resetPost` before `resetEditorBlocks` (so that meta values are available for applying)

Reply:
At some point, we should also look to refactor this so that the editor module doesn't need to store a copy of the "current" post, but instead calls to the `@wordpress/core-data` module to apply the edits on the canonical post object.

```js
wp.data.select( 'core' ).getEntityRecord( 'postType', 'post', wp.data.select( 'core/editor' ).getCurrentPostId() );
```